src/modules/sign/hooks.ts

```diff
@@ -0,0 +1,43 @@
+import { useContext, useEffect, useState } from 'react';
+import { GenericExtrinsicPayload } from '@polkadot/types';
+
+import { ExtrinsicPayloadLatestVersion } from 'constants/chainData';
+import { NetworksContext } from 'stores/NetworkContext';
+import { RegistriesContext } from 'stores/RegistriesContext';
+
+export function usePayloadDetails(
+	rawPayload: Uint8Array | string | null,
+	networkKey: string
+): [boolean, GenericExtrinsicPayload | null] {
+	const [payload, setPayload] = useState<GenericExtrinsicPayload | null>(null);
+	const [isProcessing, setIsProcessing] = useState<boolean>(false);
+	const { networks } = useContext(NetworksContext);
+	const { getTypeRegistry } = useContext(RegistriesContext);
+
+	useEffect(() => {
+		setIsProcessing(true);
+		if (getTypeRegistry === null) return;
```

Comment:
The reason for this is because in the `catch` it is returning a null value in `useRegistriesStore` so I am assuming that is there to possibly avoid complaints from the typescript compiler.  but also too, if there is issues with the network's metadata it also returns null so I would assume its a way to sanity check any errors.